server/sonar-web/src/main/js/app/components/nav/global/GlobalNavSearchForm.js

```diff
@@ -0,0 +1,350 @@
+        projects: { ...state.projects, ...keyBy(response.projects, 'key') },
+        results,
+        selected: list.length > 0 ? list[0].key : null
+      }));
+    });
+  };
+
+  searchMore = (qualifier: string) => {
+    this.setState({ loading: true, loadingMore: qualifier });
+    const recentlyBrowsed = RecentHistory.get().map(component => component.key);
+    getSuggestions(this.state.query, recentlyBrowsed, qualifier).then(response => {
+      const group = response.results.find(group => group.q === qualifier);
+      const moreResults = (group ? group.items : []).map(item => ({ ...item, qualifier }));
+      this.setState(state => ({
+        loading: false,
+        loadingMore: null,
+        more: { ...state.more, [qualifier]: 0 },
+        organizations: { ...state.organizations, ...keyBy(response.organizations, 'key') },
+        projects: { ...state.projects, ...keyBy(response.projects, 'key') },
+        results: {
+          ...state.results,
+          [qualifier]: uniqBy([...state.results[qualifier], ...moreResults], 'key')
+        }
+      }));
+    });
+  };
+
+  handleQueryChange = (event: { currentTarget: HTMLInputElement }) => {
+    const query = event.currentTarget.value;
+    this.setState({ query });
+    if (query.length === 0) {
+      this.fetchFavorites();
+    } else if (query.length >= 2) {
+      this.search(query);
+    }
+  };
+
+  selectPrevious = () => {
+    this.setState((state: State) => {
+      const list = this.getPlainComponentsList(state.results);
+      const index = list.findIndex(component => component.key === state.selected);
+      return index > 0 ? { selected: list[index - 1].key } : undefined;
+    });
+  };
+
+  selectNext = () => {
+    this.setState((state: State) => {
+      const list = this.getPlainComponentsList(state.results);
+      const index = list.findIndex(component => component.key === state.selected);
+      return index >= 0 && index < list.length - 1 ? { selected: list[index + 1].key } : undefined;
+    });
+  };
+
+  openSelected = () => {
+    if (this.state.selected) {
+      this.context.router.push(getProjectUrl(this.state.selected));
+      this.props.onClose();
+    }
+  };
+
+  scrollToSelected = () => {
+    if (this.state.selected) {
+      const node = this.nodes[this.state.selected];
+      if (node) {
+        scrollToElement(node, { topOffset: 30, bottomOffset: 30, parent: this.node });
+      }
+    }
+  };
+
+  handleKeyDown = (event: KeyboardEvent) => {
+    switch (event.keyCode) {
+      case 13:
+        event.preventDefault();
+        this.openSelected();
+        return;
+      case 27:
+        event.preventDefault();
+        this.props.onClose();
+        return;
+      case 38:
+        event.preventDefault();
+        this.selectPrevious();
+        return;
+      case 40:
+        event.preventDefault();
+        this.selectNext();
+        return;
+    }
+  };
+
+  handleSelect = (selected: string) => {
```

Comment:
I mean, you could just do `handleSelect = (selected: string) => this.setState({ selected });` like you did for `openSearch` or `closeSearch`.